src/cpp/cpp_util.cpp

```diff
@@ -11,12 +11,12 @@
 #include <util/std_expr.h>
 #include <util/symbol.h>
 
-exprt cpp_symbol_expr(const symbolt &symbol)
+symbol_exprt cpp_symbol_expr(const symbolt &symbol)
 {
   symbol_exprt tmp(symbol.name, symbol.type);
 
   if(symbol.is_lvalue)
     tmp.set(ID_C_lvalue, true);
 
-  return std::move(tmp);
+  return tmp;
```

Comment:
Why is the `std::move` being removed here?

Reply:
Perhaps counterintuitively, the std::move _prevents_ a move here. I.e., removing it enables copy elision.

Reply:
Why would that be the case? Notably, clang-7 had complained about the absence of `std::move` here, and it has just been introduced in 72bfc352035.

Reply:
@tautschnig the key is whether a type conversion is required or not. Previously we returned a `symbol_exprt`, meaning we needed a conversion to `exprt` (via its copy- or move-constructor), so `std::move` was a good thing (picking the move constructor). Now we return a `symbol_exprt`, meaning no type conversion is necessary and RVO / copy/move-elision is possible, so `std::move` is now a *bad* thing.

Reply:
Many thanks for the elaboration!